js/bitmex.js

```diff
@@ -259,6 +260,130 @@ module.exports = class bitmex extends Exchange {
         });
     }
 
+    async fetchCurrencies (params = {}) {
+        /**
+         * @method
+         * @name bitmex#fetchCurrencies
+         * @description fetches all available currencies on an exchange
+         * @param {object} params extra parameters specific to the bitmex api endpoint
+         * @returns {object} an associative dictionary of currencies
+         */
+        const assets = await this.publicGetWalletAssets (params);
+        //
+        //    [
+        //        {
+        //            asset: 'XBT',
+        //            currency: 'XBt',
+        //            majorCurrency: 'XBT',
+        //            name: 'Bitcoin',
+        //            currencyType: 'Crypto',
+        //            scale: '8',
+        //            enabled: true,
+        //            isMarginCurrency: true,
+        //            minDepositAmount: '10000',
+        //            minWithdrawalAmount: '1000',
+        //            maxWithdrawalAmount: '100000000000000',
+        //            networks: [
+        //                {
+        //                    asset: 'btc',
+        //                    tokenAddress: '',
+        //                    depositEnabled: true,
+        //                    withdrawalEnabled: true,
+        //                    withdrawalFee: '20000',
+        //                    minFee: '20000',
+        //                    maxFee: '10000000'
+        //                }
+        //            ]
+        //        },
+        //        ...
+        //    ]
+        //
+        const result = {};
+        for (let i = 0; i < assets.length; i++) {
+            const currency = assets[i];
+            const currencyId = this.safeString (currency, 'asset');
+            const code = this.safeCurrencyCode (currencyId);
+            const scale = this.safeString2 (currency, 'scale');
+            const precision = this.parsePrecision (scale);
```

Comment:
@kroitor @carlosmiei @frosty00 Do you think this is correct? I saw this done on Ascendex, but previously the precision for BTC is 1e-10, and after this PR it's 1e-8